edb/server/main.py

```diff
@@ -69,6 +71,26 @@
 _server_initialized = False
 
 
+class FileModifiedHandler(watchdog.events.FileSystemEventHandler):
+    def __init__(self, reload_certs: Callable[[], None]):
+        self._reload_certs = reload_certs
+
+    def on_modified(
+        self,
+        event: Union[
+            watchdog.events.DirModifiedEvent,
+            watchdog.events.FileModifiedEvent,
+        ]
+    ) -> None:
+        if isinstance(event, watchdog.events.DirModifiedEvent):
+            return
+
+        try:
+            self._reload_certs()
```

Comment:
This is incorrect - you shouldn't call methods on Server from an arbitrary thread. You should schdule them with `Server._loop.call_soon_threadsafe`. Also need to shutdown the server if any error happens during reading the files or modifying the tls context.

Reply:
> Also need to shutdown the server if any error happens during reading the files or modifying the tls context.

Is using the original cert not an option if the new one is invalid?

Reply:
I have a feeling we should abort in that case, but it's not a strong one. If we decide we want to abort, we should at least retry reading after some time and only abort on a second time. @elprans thoughts?

Reply:
I added a fast exponential retry & abort.